server/src/service.rs

```diff
@@ -293,4 +297,57 @@ impl Service {
 
         Ok(())
     }
+
+    pub async fn set_locked(
+        &self,
+        locked: bool,
+        objects: &[OwnedObjectPath],
+    ) -> Result<(Vec<OwnedObjectPath>, Vec<OwnedObjectPath>), ServiceError> {
+        let mut locked_without_prompt = Vec::new();
```

Comment:
maybe drop the locked from the names, otherwise with_prompt/without_prompt are decent :)

Reply:
Done.